sdk/storage/src/queue/clients/queue_service_client.rs

```diff
@@ -1,59 +1,56 @@
 use crate::core::Client;
+use crate::queue::clients::QueueNameClient;
 use crate::requests;
-use crate::{HasStorageClient, IntoQueueServiceClient, QueueService, WithQueueServiceClient};
-use std::borrow::Cow;
+use crate::HasStorageClient;
 use std::fmt::Debug;
 
 #[derive(Debug, Clone)]
-pub struct QueueServiceClient<'a, C>
+pub struct QueueServiceClient<C>
```

Comment:
Yes the names can be better. The idea is to have a client for the "topmost" queue functions (ie the one without a queue name, like `list_queues`) and one for the functions that require a queue name (akin as cosmos -> database -> collection).

So for example one will insert a message by creating a blob storage client (which holds the security info). Then will converted to *queue client*. And finally, by specifying a queue name they can insert messages.

Maybe `QueueAccountClient` and `QueueClient` are better fit?

Reply:
Those could work. I wonder for this particular case if it makes more sense for this to just be one client where all the methods that require a name, simply take that name. Following the same pattern from Cosmos feels a bit forced as queues don't follow the same resources model as cosmos does.

Reply:
Done! Honestly I think it's useful to avoid repeating the queue name over and over. 
Also, if you ever want you can get the queue account client back calling the getter function.

The only gripe is the need to clone (not that I think I will happen often). If need be we could use ref counting (but I agree with you, cloning something is not that bad).